weather.ozweather/resources/lib/abcvideo.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+
+# *  This Program is free software; you can redistribute it and/or modify
+# *  it under the terms of the GNU General Public License as published by
+# *  the Free Software Foundation; either version 2, or (at your option)
+# *  any later version.
+# *
+# *  This Program is distributed in the hope that it will be useful,
+# *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+# *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# *  GNU General Public License for more details.
+# *
+# *  You should have received a copy of the GNU General Public License
+# *  along with KODI; see the file COPYING. If not, write to
+# *  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+# *  http://www.gnu.org/copyleft/gpl.html
+# *
+
+import requests
+import re
+
+try:
+    from .common import log as log
+except ImportError:
+    print("\nKodi is not available -> probably unit testing")
```

Comment:
Should this be used after the function declaration (it is right below)? Use log instead of print as you are "aliasing" in the method below.

Reply:
So this is the only thing I am a bit stuck on, I think.

I really want to be able to keep unit testing these sub-modules (abc, bom, weatherzone .py) - because it makes dealing with required upstream changes (i.e. scraping changes) _very_ much easier and quicker.  Hence this little bit of code basically will raise the exception if the module is run outside of Kodi and then, in only that case, defines a log function (that is just print).

For the moment I'll add an explanatory comment above this bit of code, but more than happy to accept suggestions for a better approach that lets me maintain the unit testing capability but is better for Kodi and/or more pythonic

(In case you haven't guessed, I am a Python hacker at best!)

Reply:
that's fine